demo/src/main/java/com/example/demo/raport/RaportController.java

```diff
@@ -0,0 +1,104 @@
+package com.example.demo.raport;
+
+import com.example.demo.program.Program;
+import com.example.demo.program.ProgramRepository;
+import com.example.demo.programmer.Programmer;
+import com.example.demo.programmer.ProgrammerRepository;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.transaction.annotation.Transactional;
+import org.springframework.web.bind.annotation.*;
+
+import java.sql.Timestamp;
+import java.util.List;
+import java.util.Optional;
+
+@RestController
+@CrossOrigin
+@RequestMapping("/raport")
+public class RaportController {
+    private final RaportRepository raportRepository;
+    private final ProgramRepository programRepository;
+    private final ProgrammerRepository programmerRepository;
+
+    public RaportController(RaportRepository raportRepository, ProgramRepository programRepository, ProgrammerRepository programmerRepository) {
+        this.raportRepository = raportRepository;
+        this.programRepository = programRepository;
+        this.programmerRepository = programmerRepository;
+    }
+    @Transactional
+    @GetMapping("/getAll")
+    public ResponseEntity<List<Raport>>getAll(){
+        List<Raport> result = raportRepository.findAll();
+        if(result.isEmpty()){
```

Comment:
jak dla mnie to ten kod do tego nie służy. Poza tym jak nie ma tych raportów to zwrócenie pustej listy też jest poprawnym wynikiem.

Reply:
Poprawione